_posts/2020-07-31-fnn-vae-for-noisy-timeseries/fnn_vae_for_noisy_timeseries.Rmd

```diff
@@ -0,0 +1,509 @@
+Training goes a *lot* faster for FNN-VAE.
+
+And secondly, while we don't see much difference between FNN-LSTM and FNN-VAE, we *do* see a clear impact of using FNN loss:
+Especially for the VAE architecture, but for the LSTM one as well, adding in FNN loss strongly reduces mean squared error with
+respect to the underlying (denoised) series. This is of particular interest with VAE, as it comes with a regularizer
+out-of-the-box -- namely, Kullback-Leibler (KL) divergence.
+
+Of course, we don't claim that similar results will always be obtained on other noisy series; and neither did we tune any of
+the models "to death". For, what could be the intent of such a post but to show our readers interesting (and promising) ideas
+to pursue in their own experimentation?
+
+## The context
+
+This post is the third in a mini-series.
+
+In [Deep attractors: Where deep learning meets chaos](https://blogs.rstudio.com/ai/posts/2020-06-24-deep-attractors/), we
+explained, with a substantial detour into chaos theory, the idea of FNN loss, introduced in [@gilpin2020deep]. Please consult
+that first post for theoretical background and intuitions behind the technique.
+
+The subsequent post, [Time series prediction with FNN-LSTM](https://blogs.rstudio.com/ai/posts/2020-07-20-fnn-lstm/), showed
+how to use it for prediction (as opposed to reconstructing an attractor). "It", the technique, meant augmenting an LSTM
+autoencoder with FNN loss. The results were stunning: In multi-step prediction (12-120 steps, with that number varying by
+dataset), the short-term forecasts were drastically improved by adding in FNN regularization. See that second post for
+experimental setup and results on four very different, non-synthetic datasets.
+
+Today, we show how to replace the LSTM autoencoder by a -- convolutional -- VAE. In light of the experimentation results,
+already hinted at above, it is completely thinkable that the "variational" part is not even so important here -- that a
+convolutional autoencoder with just MSE loss would have performed just as well on those data. In fact, to find out, it's
+enough to just remove the call to `reparameterize()` and multiply the KL component of the loss by 0. (We leave this to the
+interested reader, to keep the post at reasonable length.)
+
+One last contextual thing, in case you haven't read the two previous posts and would like to jump in here directly. We're
+doing time series forecasting; so why this talk of autoencoders? Shouldn't we just be comparing an LSTM (or some other type of
+RNN, for that matter) to a convnet? In fact, the necessity of a latent representation is due to the very idea of FNN: The
+latent code is supposed to reflect the true attractor of a dynamical system. That is, if the attractor of the underlying
+system is roughly two-dimensional, we hope to find that just two of the latent variables have considerable variance. (This
```

Comment:
Really nicely done! That's one heck of a thing to describe, but the sentence flows very nicely.